src/librustc_mir/transform/const_prop.rs

```diff
@@ -857,22 +880,32 @@ impl<'mir, 'tcx> MutVisitor<'tcx> for ConstPropagator<'mir, 'tcx> {
                 if let Some(local) = place.as_local() {
                     let can_const_prop = self.can_const_prop[local];
                     if let Some(()) = self.const_prop(rval, place_layout, source_info, place) {
-                        if can_const_prop == ConstPropMode::FullConstProp
-                            || can_const_prop == ConstPropMode::OnlyPropagateInto
-                        {
+                        if can_const_prop != ConstPropMode::NoPropagation {
+                            // This will return None for Locals that are from other blocks,
+                            // so it should be okay to propagate from here on down.
                             if let Some(value) = self.get_const(local) {
                                 if self.should_const_prop(value) {
                                     trace!("replacing {:?} with {:?}", rval, value);
                                     self.replace_with_const(rval, value, statement.source_info);
-
                                     if can_const_prop == ConstPropMode::FullConstProp {
+                                        // FIXME(felix91gr): is this trace accurate??
```

Comment:
I think this FIXME can be removed

Reply:
If you're thinking about the one we talked about before, that one was another fixme, and it's already been deleted :sparkles: 

But regarding this one... is it actually accurate? It seems to me that the if should either disappear, or allow for more than just `FullConstProp`

Reply:
Oh, I see what you're saying. Yeah it should also include the new mode. Good point!